src/layout/umap.c

```diff
@@ -92,27 +150,82 @@ static igraph_error_t igraph_i_umap_find_sigma(const igraph_vector_t *distances,
 }
 
 
-/* Convert the graph with distances into a probability graph with exponential decay */
-/* NOTE: this is funny because the distance was a correlation to start with... ?? */
-static igraph_error_t igraph_i_umap_find_prob_graph(const igraph_t *graph,
-        const igraph_vector_t *distances, igraph_vector_t *umap_weights) {
+/**
+ * \function igraph_layout_umap_compute_weights
+ * \brief Compute weights for a UMAP layout starting from distances.
+ *
+ * \experimental
+ *
+ * UMAP is used to embed high-dimensional vectors in a low-dimensional space
+ * (most commonly 2D). It uses a distance graph as an intermediate data structure,
+ * making it also a useful graph layout algorithm. See \ref igraph_layout_umap() for more information.
+ *
+ * </para><para>
+ *
+ * An early step in UMAP is to compute exponentially decaying "weights" from the
+ * distance graph. Connectivities can also be viewed as edge weights that quantify
+ * similarity between two vertices. This function computes weights from the
+ * distance graph. To compute the layout from precomputed weights, call
+ * igraph_layout_umap with the "distances_are_weights" argument set to true.
+ *
+ * </para><para>
+ *
+ * Technical note: For each vertex, this function computes its scale factor (sigma),
+ * its connectivity correction (rho), and finally the weights themselves.
+ *
+ * </para><para>
+ * References:
+ *
+ * </para><para>
+ * Leland McInnes, John Healy, and James Melville. https://arxiv.org/abs/1802.03426
+ *
+ * \param graph Pointer to the distance graph.
+ * \param distances Pointer to the vector with the vertex-to-vertex distance associated with
+ *   each edge.
```

Comment:
Can you please mention that this is allowed to be `\c NULL`?

Reply:
done